pkg/R/pairRDD.R

```diff
@@ -769,4 +769,84 @@ setMethod("sortByKey",
             newRDD <- partitionBy(x, numPartitions, rangePartitionFunc)
             lapplyPartition(newRDD, partitionFunc)
           })
-          
+
+#' @description
+#' \code{sampleByKey} return a subset RDD of the given RDD sampled by key 
+#'
+#' @param x The RDD to sample elements by key, where each element is
+#'             list(K, V) or c(K, V). 
+#' @param withReplacement Sampling with replacement or not
+#' @param fraction The (rough) sample target fraction
+#' @param seed Randomness seed value
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, 1:2000)
+#' makePairs <- lapply(rdd, function(x) { if(x%%2 == 0) list("a", x) else list("b", x) })
+#' fractions <- list(c("a", 0.2), c("b", 0.1))
```

Comment:
What do you think about this interface change: let `fractions` just be a named list. E.g.:
```
fractions <- list(a = 0.2, b = 0.1)
```
This should be easier to use for users.